src/Workspaces/Core/MSBuild.BuildHost/Microsoft.CodeAnalysis.Workspaces.MSBuild.BuildHost.csproj

```diff
@@ -20,19 +20,32 @@
          a RuntimeIdentifier given to us, since we really don't want to accidentally pick up platform specific binaries since we also have to run on Mono.
          -->
     <PlatformTarget>AnyCPU</PlatformTarget>
+
+    <!--
+      We specifically reference Microsoft.Build 17.3.4, since it's the oldest version still compatible $(NetRoslynBuildHostNetCoreVersion).
+      The version targeted is only used for build time, since we use MSBuildLocator to discover the proper version at runtime.
+    -->
+    <_MsbuildVersion>17.3.4</_MsbuildVersion>
   </PropertyGroup>
+  <ItemGroup Condition="'$(DotnetBuildFromSource)' == 'true'">
```

Comment:
Is there a reason we wouldn't just do the PackageDownloadAndReference for source build too? I guess maybe it's going to break source build because it doesn't know how to redirect it to the source built version?

Reply:
Sounds good.